common/step_download_test.go

```diff
@@ -188,10 +187,10 @@ func TestStepDownload_Run(t *testing.T) {
 			},
 		},
 		{"successfull absolute symlink - checksum from parameter - checksum type",
-			fields{Extension: "txt", Url: []string{abs(t, "./test-fixtures/root/another.txt") + "?"}, ChecksumType: "sha1", Checksum: cs["/root/another.txt"]},
+			fields{Extension: "txt", Url: []string{abs(t, "./test-fixtures/root/another.txt") + "?"}, Checksum: "sha1:" + cs["/root/another.txt"]},
 			multistep.ActionContinue,
 			[]string{
-				toSha1(cs["/root/another.txt"]) + ".txt.lock",
+				toSha1("sha1:"+cs["/root/another.txt"]) + ".txt.lock",
 			},
```

Comment:
It would be cool to add some tests here that check that we can correctly infer common checksum types, in case the type is omitted.

Reply:
We already do; the cases that have `Checksum: "sha1:" +` actually set it and the other ones don't